src/v/datalake/datalake_manager.cc

```diff
@@ -263,15 +263,17 @@ ss::future<> datalake_manager::start() {
 }
 
 ss::future<> datalake_manager::stop() {
+    vlog(datalake_log.debug, "Stopping datalake manager...");
     auto f = _gate.close();
     co_await _backlog_controller->stop();
     _deregistrations.clear();
     co_await ss::max_concurrent_for_each(
       _translators, 32, [](auto& entry) mutable {
           return entry.second->stop();
       });
-    co_await std::move(f);
     _schema_cache->stop();
+    co_await std::move(f);
```

Comment:
I don't understand what changes by swapping these 2 lines. The `gate.close()` task is already running since scheduled on line 267. Whether or not you waited on the future to resolve does not have an influence of how the underlying state looks except that now stop() method is called slightly faster and possibly you are just masking a race condition?

Reply:
Ah that's fair. I think I went overboard here thinking it's never safe to access state outside of a gate. But in a stop() method it's probably fine. Will remove this aspect of this PR

Reply:
Removed, and squashed the remaining logging-only commits together